src/main/java/frc/robot/subsystems/scoring/ShooterIOSim.java

```diff
@@ -3,23 +3,35 @@
 import edu.wpi.first.math.controller.PIDController;
 import edu.wpi.first.math.controller.SimpleMotorFeedforward;
 import edu.wpi.first.math.system.plant.DCMotor;
+import edu.wpi.first.wpilibj.DigitalInput;
 import edu.wpi.first.wpilibj.simulation.FlywheelSim;
 import frc.robot.Constants;
-import frc.robot.Constants.Scoring;
+import frc.robot.Constants.ConversionConstants;
+import frc.robot.Constants.ScoringConstants;
 
 public class ShooterIOSim implements ShooterIO {
     // TODO: Tune this later
-    private final FlywheelSim shooterSim = new FlywheelSim(DCMotor.getNeoVortex(1), 1.0, 1.0);
-    private final PIDController shooterController = new PIDController(Scoring.shooterkP, Scoring.shooterkI, Scoring.shooterkD);
-    private final SimpleMotorFeedforward shooterFeedforward = new SimpleMotorFeedforward(Scoring.shooterkS, Scoring.shooterkV, Scoring.shooterkA);
+    private final FlywheelSim shooterSim = new FlywheelSim(DCMotor.getNeoVortex(1), 1.0, 0.1);
+    private final PIDController shooterController =
+            new PIDController(
+                    ScoringConstants.shooterkP,
+                    ScoringConstants.shooterkI,
+                    ScoringConstants.shooterkD);
+    private final SimpleMotorFeedforward shooterFeedforward =
+            new SimpleMotorFeedforward(
+                    ScoringConstants.shooterkS,
+                    ScoringConstants.shooterkV,
+                    ScoringConstants.shooterkA);
+
+    DigitalInput bannerSensor = new DigitalInput(Constants.SensorConstants.bannerPort);
```

Comment:
Is DigitalInput able to be simulated directly?

It might make sense to have a Constants.java constant for simulation override that the scoring subsystem can use to ignore any logic if we have a note or not

Reply:
Currently, you can control the banner sensor to pretend that the beam was broken in the WPI sim dashboard.

Reply:
That works for now - I'll create a ticket to toggle override since it could be useful in a match (low-ish priority though)